src/soroban/assembled_transaction.ts

```diff
@@ -0,0 +1,717 @@
+import type { ContractClientOptions, XDR_BASE64 } from ".";
+import {
+  Account,
+  BASE_FEE,
+  Contract,
+  Operation,
+  SorobanRpc,
+  StrKey,
+  TimeoutInfinite,
+  TransactionBuilder,
+  authorizeEntry,
+  hash,
+  xdr,
+} from "..";
+import type { Memo, MemoType, Transaction } from "..";
+
+type Tx = Transaction<Memo<MemoType>, Operation[]>;
+
+type SendTx = SorobanRpc.Api.SendTransactionResponse;
+type GetTx = SorobanRpc.Api.GetTransactionResponse;
+
+/**
+ * Error interface containing the error message
+ */
+interface ErrorMessage {
+  message: string;
+}
+
+interface Result<T, E extends ErrorMessage> {
+  unwrap(): T;
+  unwrapErr(): E;
+  isOk(): boolean;
+  isErr(): boolean;
+}
+class Ok<T> implements Result<T, never> {
+  constructor(readonly value: T) {}
+  unwrapErr(): never { throw new Error("No error") }
+  unwrap() { return this.value }
+  isOk() { return true }
+  isErr() { return false }
+}
+class Err<E extends ErrorMessage> implements Result<never, E> {
+  constructor(readonly error: E) {}
+  unwrapErr() { return this.error }
+  unwrap(): never { throw new Error(this.error.message) }
+  isOk() { return false }
+  isErr() { return true }
+}
```

Comment:
Is creating these Rust-isms necessary? It doesn't feel right.

Reply:
I meant to add (and will add) TS Doc info above these, sorry for the confusion. This again is not new, and whether we nailed the API is an open question. 

Most contract methods I've seen in examples return simple values, not Results. They might also panic. To represent these, we parse the result to the corresponding JS type, or throw an error.

For contract methods that return Rust `Result` types, do we want the same behavior? Maybe. "Value or error" is a much more common JS pattern than anything like a `Result` type. But then we lose information. The `Result` return types end up having the same JS treatment as the non-Results. So maybe it's worth having some minimal Result implementation, as done here.

We've taken care to not clutter the global export namespace with `Ok` and `Err`. They are only exposed at `AssembledTransaction.Result.(Ok|Err)`. With good TS Doc, I think it's a reasonable starting implementation. It at least spurs better questions and conversation than simplifying Rust Results down to the same "value or error" implementation as all other methods.

Reply:
I've added TS Doc. Leaving in for now.